src/test/java/seedu/address/storage/JsonOrderBookStorageTest.java

```diff
@@ -0,0 +1,95 @@
+import seedu.address.model.OrderBook;
+import seedu.address.model.ReadOnlyOrderBook;
+
+
+public class JsonOrderBookStorageTest {
+    private static final Path TEST_DATA_FOLDER = Paths.get("src", "test", "data", "JsonOrderBookStorageTest");
+
+    @TempDir
+    public Path testFolder;
+
+    @Test
+    public void readOrderBook_nullFilePath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> readOrderBook(null));
+    }
+
+    private java.util.Optional<ReadOnlyOrderBook> readOrderBook(String filePath) throws Exception {
+        return new JsonOrderBookStorage(Paths.get(filePath)).readOrderBook(addToTestDataPathIfNotNull(filePath));
+    }
+
+    private Path addToTestDataPathIfNotNull(String prefsFileInTestDataFolder) {
+        return prefsFileInTestDataFolder != null
+                ? TEST_DATA_FOLDER.resolve(prefsFileInTestDataFolder)
+                : null;
+    }
+
+    @Test
+    public void readOrderBook_missingFile_emptyResult() throws Exception {
+        assertFalse(readOrderBook("NonExistentFile.json").isPresent());
+    }
+
+    @Test
+    public void readOrderBook_notJsonFormat_exceptionThrown() {
+        assertThrows(DataConversionException.class, () -> readOrderBook("notJsonFormatOrderBook.json"));
+    }
+
+    @Test
+    public void readOrderBook_invalidOrderOrderBook_throwDataConversionException() {
+        assertThrows(DataConversionException.class, () -> readOrderBook("invalidOrderOrderBook.json"));
+    }
+
+    @Test
+    public void readOrderBook_invalidAndValidOrdersOderBook_throwDataConversionException() {
+        assertThrows(DataConversionException.class, () -> readOrderBook("invalidAndValidOrdersOrderBook.json"));
+    }
+
+    @Test
+    public void readAndSaveOrderBook_allInOrder_success() throws Exception {
+        Path filePath = testFolder.resolve("TempOrderBook.json");
+        OrderBook original = getTypicalOrderBook();
+        JsonOrderBookStorage jsonOrderBookStorage = new JsonOrderBookStorage(filePath);
+
+        // Save in new file and read back
+        jsonOrderBookStorage.saveOrderBook(original, filePath);
+        ReadOnlyOrderBook readBack = jsonOrderBookStorage.readOrderBook(filePath).get();
+        assertEquals(original, new OrderBook(readBack));
+
+    }
+
+    @Test
+    public void saveOrderBook_nullOrderBook_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> saveOrderBook(null, "SomeFile.json"));
+    }
+
+    /**
+     * Saves {@code orderBook} at the specified {@code filePath}.
+     */
+    private void saveOrderBook(ReadOnlyOrderBook orderBook, String filePath) {
+        try {
+            new JsonOrderBookStorage(Paths.get(filePath))
+                    .saveOrderBook(orderBook, addToTestDataPathIfNotNull(filePath));
+        } catch (IOException ioe) {
+            throw new AssertionError("There should not be an error writing to the file.", ioe);
+        }
+    }
```

Comment:
I'm not sure why you re-implemented saveOrderbook here, instead of using the one from the original file? Just checking if this is intentional.

Reply:
This was the way the address book originally implemented the storage test, so I guess having this is fine?